src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeServiceClient.java

```diff
@@ -0,0 +1,226 @@
+   */
+  DropChannelResponse dropChannel(DropChannelRequestInternal request) {
+    DropChannelResponse response =
+        executeApiRequestWithRetries(
+            request,
+            DropChannelResponse.class,
+            DROP_CHANNEL_ENDPOINT,
+            "drop channel",
+            STREAMING_DROP_CHANNEL,
+            ErrorCode.DROP_CHANNEL_FAILURE);
+
+    if (response.getStatusCode() != RESPONSE_SUCCESS) {
+      logger.logDebug(
+          "Drop channel request failed, table={}, message={}",
+          request.getFullyQualifiedTableName(),
+          response.getMessage());
+      throw new SFException(ErrorCode.DROP_CHANNEL_FAILURE, response.getMessage());
+    }
+    return response;
+  }
+
+  /**
+   * Gets the status of a channel given a {@link ChannelsStatusRequest}.
+   *
+   * @param request the channel status request
+   * @return the response from the channel status request
+   */
+  ChannelsStatusResponse channelStatus(ChannelsStatusRequest request) {
+    ChannelsStatusResponse response =
+        executeApiRequestWithRetries(
+            request,
+            ChannelsStatusResponse.class,
+            CHANNEL_STATUS_ENDPOINT,
+            "channel status",
+            STREAMING_CHANNEL_STATUS,
+            ErrorCode.CHANNEL_STATUS_FAILURE);
+
+    if (response.getStatusCode() != RESPONSE_SUCCESS) {
+      logger.logDebug("Channel status request failed, message={}", response.getMessage());
+      throw new SFException(ErrorCode.CHANNEL_STATUS_FAILURE, response.getMessage());
+    }
+    return response;
+  }
+
+  /**
+   * Registers a blob given a {@link RegisterBlobRequest}.
+   *
+   * @param request the register blob request
+   * @param executionCount the number of times the request has been executed, used for logging
+   * @return the response from the register blob request
+   */
+  RegisterBlobResponse registerBlob(RegisterBlobRequest request, final int executionCount) {
+    RegisterBlobResponse response =
+        executeApiRequestWithRetries(
+            request,
+            RegisterBlobResponse.class,
+            REGISTER_BLOB_ENDPOINT,
+            "register blob",
+            STREAMING_REGISTER_BLOB,
+            ErrorCode.REGISTER_BLOB_FAILURE);
+
+    if (response.getStatusCode() != RESPONSE_SUCCESS) {
+      logger.logDebug(
+          "Register blob request failed for blob={}, message={}, executionCount={}",
+          request.getBlobs().stream().map(BlobMetadata::getPath).collect(Collectors.toList()),
+          response.getMessage(),
+          executionCount);
+      throw new SFException(ErrorCode.REGISTER_BLOB_FAILURE, response.getMessage());
+    }
+    return response;
+  }
+
+  private <T extends StreamingIngestResponse> T executeApiRequestWithRetries(
+      StreamingIngestRequest request,
+      Class<T> responseClass,
+      String endpoint,
+      String operation,
+      ServiceResponseHandler.ApiName apiName,
+      ErrorCode errorCode) {
```

Comment:
unclear what this errorcode is doing in the request signature. mind adding a javadoc to the method ?

Reply:
ok i see why. Please remove the catch block from this method and let all callers do their own catch + rethrow with whatever errorcode they need.

Reply:
Moved the try catch clause to call sites.